modules/hal_nordic/nrfx/nrfx_glue.h

```diff
@@ -381,7 +381,7 @@ void nrfx_busy_wait(uint32_t usec_to_wait);
 #define NRFX_PPI_GROUPS_USED_BY_802154_DRV     0
 #endif // CONFIG_NRF_802154_RADIO_DRIVER
 
-#if defined(CONFIG_NRF_802154_RADIO_DRIVER) && !defined(CONFIG_NRF_802154_SL_OPENSOURCE)
+#if defined(CONFIG_MPSL) && !defined(CONFIG_NRF_802154_SL_OPENSOURCE)
```

Comment:
Shouldn't the `&& !defined(CONFIG_NRF_802154_SL_OPENSOURCE)` part be actually dropped?

Reply:
I don't know, do you know anyone who works on SL_OPENSOURCE that is able to comment?

Reply:
I think the intention of having
```c
#if defined(CONFIG_NRF_802154_RADIO_DRIVER) && !defined(CONFIG_NRF_802154_SL_OPENSOURCE)
```
is to detect the MPSL in downstream while not explicit referencing it here.
So agree with @anangl, only
```c
#if defined(CONFIG_MPSL)
```
should be left.